src/lib/index.js

```diff
@@ -1076,7 +1078,9 @@ lib.hovertemplateString = function(string, labels) {
         }
 
         if(format) {
-            value = d3.format(format.replace(TEMPLATE_STRING_FORMAT_SEPARATOR, ''))(value);
+            var fmt = d3format;
+            if(locale) fmt = fmt.formatLocale(locale);
+            value = fmt.format(format.replace(TEMPLATE_STRING_FORMAT_SEPARATOR, ''))(value);
```

Comment:
this seems a little funny - can you just use `fullLayout._d3locale` and avoid pulling in a new complete copy of `d3-format`? Anyway that's going to have different capabilities from the one in d3v3 that we already have and use, which will be confusing to users.

Reply:
> this seems a little funny - can you just use `fullLayout._d3locale` and avoid pulling in a new complete copy of `d3-format`

Nice catch :eye: ! I agree this was a bad idea. In 1faaf8b6c3c161e243ed32e2bbf139c678b135ea, I reuse our existing formatting system.